tools/c7n_azure/c7n_azure/actions/lock.py

```diff
@@ -0,0 +1,66 @@
+from c7n_azure.utils import ResourceIdParser
+
+from c7n.utils import type_schema
+from c7n_azure.actions.base import AzureBaseAction
+from azure.mgmt.resource.locks import ManagementLockClient
+from azure.mgmt.resource.locks.models import ManagementLockObject
+
+
+class LockAction(AzureBaseAction):
+
+    """
+      Perform lock operation on any ARM resource. Can be used with
+      generic resource type `armresource` or on any other more specific
+      ARM resource type supported by Cloud Custodian.
+
+      Lock can be of 2 types: ReadOnly and CanNotDelete. Lock type is required.
+
+      To create or delete management locks, you must have proper access.
+      See [Who can create or delete locks](https://docs.microsoft.com/en-us/azure/
+      azure-resource-manager/resource-group-lock-resources#who-can-create-or-delete-locks)
+
+      Example of the policy that adds ReadOnly lock to all keyvaults:
+
+     .. code-block:: yaml
+
+       policies:
+          - name: lock-keyvaults
+            resource: azure.keyvault
+            actions:
+              - type: lock
+                lock-type: ReadOnly
+     """
+
+    schema = type_schema(
+        'lock',
+        required=['lock-type'],
+        **{
+            'lock-type': {'type': 'string'}
```

Comment:
please use an enum so its a bit more obvious what the supported values are (including disabling a lock).